service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java

```diff
@@ -186,6 +194,85 @@ public List<CopyNumberCountByGene> getCNAAlterationCountByGenes(List<String> stu
+        
+        List<MolecularProfile> molecularProfiles = molecularProfileService.getMolecularProfilesInStudies(studyIds,
+            "SUMMARY");
+
+        Map<String, List<MolecularProfile>> molecularProfileMap = molecularProfileUtil
+            .categorizeMolecularProfilesByStableIdSuffixes(molecularProfiles);
+
+        Set<String> hugoGeneSymbols = genomicDataFilters.stream().map(Pair::getKey)
+            .collect(Collectors.toSet());
+
+        Map<String, Integer> geneSymbolIdMap = geneService
+            .fetchGenes(new ArrayList<>(hugoGeneSymbols), "HUGO_GENE_SYMBOL",
+                "SUMMARY")
+            .stream().collect(Collectors.toMap(Gene::getHugoGeneSymbol, Gene::getEntrezGeneId));
+        
+        return genomicDataFilters
+            .stream()
+            .flatMap(gdFilter -> {
+                GenomicDataCountItem genomicDataCountItem = new GenomicDataCountItem();
+                String hugoGeneSymbol = gdFilter.getKey();
+                String profileType = gdFilter.getValue();
+                genomicDataCountItem.setHugoGeneSymbol(hugoGeneSymbol);
+                genomicDataCountItem.setProfileType(profileType);
+                
+                List<String> stableIds = Arrays.asList(geneSymbolIdMap.get(hugoGeneSymbol).toString());
+
+                Map<String, String> studyIdToMolecularProfileIdMap = molecularProfileMap
+                    .getOrDefault(profileType, new ArrayList<MolecularProfile>()).stream()
+                    .collect(Collectors.toMap(MolecularProfile::getCancerStudyIdentifier,
+                        MolecularProfile::getStableId));
+                
+                List<String> mappedSampleIds = new ArrayList<>();
+                List<String> mappedProfileIds = new ArrayList<>();
+
+                for (int i = 0; i < sampleIds.size(); i++) {
+                    String studyId = studyIds.get(i);
+                    if (studyIdToMolecularProfileIdMap.containsKey(studyId)) {
+                        mappedSampleIds.add(sampleIds.get(i));
+                        mappedProfileIds.add(studyIdToMolecularProfileIdMap.get(studyId));
+                    }
+                }
+
+                if (mappedSampleIds.isEmpty()) {
+                    return Stream.of();
+                }
+
+                List<GeneMolecularData> geneMolecularDataList = molecularDataService.getMolecularDataInMultipleMolecularProfiles(mappedProfileIds, mappedSampleIds,
+                        stableIds.stream().map(Integer::parseInt).collect(Collectors.toList()), "SUMMARY");
+                    
+                List<GenomicDataCount> genomicDataCounts = geneMolecularDataList
+                    .stream()
+                    .collect(Collectors.groupingBy(GeneMolecularData::getValue))
+                    .entrySet()
+                    .stream()
+                    .filter(entry -> entry.getKey() != null && !entry.getKey().equals("NA") && entry.getValue().size() > 0)
```

Comment:
are conditions specified here possible to happen?

Reply:
Yes. Sometime getKey(), aka alteration, will return "NA".

Reply:
Yes, I `NA` should be it's own category